### ISSUE TITLE
On server restart defined jobs are not scheduled

### ISSUE THREAD

**Issue by teledna** [rschmukler/agenda#280]
I have an express server which schedules jobs. If my express server goes down and restarted, jobs which are scheduled are not running

**Comment by flashsnake-so** [comment]
Did those jobs still store in database?

**Comment by teledna** [comment]
Yes,jobs are available in database

**Comment by rschmukler** [comment]
My guess is that you are defining the definitions dynamically. Definitions are not stored, only attributes and definition names. Ie.

Instead of:

```js
var userId = 123; // loaded elsewhere
agenda.define('email-user-' + userId, function(done) {
  emailUser(userId, done);
});
agenda.now('email-user-'+userId);
```

Do this:

```js
// Define this in a file that is always run (eg. require('./agenda-definitions'); 
agenda.define('email-user', function(job, done) {
  emailUser(job.attrs.userId, done);
});
```
```js
//Call this when you need to send an email
var userId = 123;  // loaded elsewhere
agenda.now('email-user', { userId: 123 })
```

Hope this helps,
Ryan

**Comment by teledna** [comment]
Thanks for immediate reply. I will change it and test

Regards
Vijaya

**Comment by teledna** [comment]
Hi,

This is sample program

var express = require('express');
var app = express();
var Agenda = require('agenda');
var fs = require('fs');
url = require('url'),
http = require('http'),
qs = require('querystring')

var logStream = fs.createWriteStream('log.txt', {'flags': 'a'});
var mongoConnectionString = "mongodb://172.16.11.155/AgendaDB";
var agenda = new Agenda({db: {address: mongoConnectionString, collection: "persist"}});


agenda.define('schedulecall', function(job, done) {
        var id = job.attrs.data.id;
        console.log("id is :"+id);
        logStream.write("Executed at :"+new Date()+" id:"+id);
        });
agenda.on('ready', function() {
        console.log("Inside on ready()");
});
app.get('/schedule', function (req, res) {
  var url_parts = url.parse(req.url,true);
  console.log(url_parts.query);
  var scheduleId = url_parts.query.id;
  var date = url_parts.query.date;
  console.log("id:"+scheduleId);
  console.log("date:"+date);
  agenda.schedule(new Date(date), 'schedulecall', {id:scheduleId });
  //agenda.every('5 seconds', 'schedulecall', {id:scheduleId });
  agenda.start();
  res.send('Hello World!');

});

app.listen(3100, function () {
  console.log('Example app listening on port 3100!');
});


I am scheduling when a request comes on /schedule. Assume that I have got two requests and I have scheduled those. My server went down before those jobs are executed. On restarting how to make sure that these jobs are rescheduled and executed


Regards
Vijaya
